src/StoreKeeper/WooCommerce/B2C/Imports/ShippingMethodImport.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\Imports;
+
+use Adbar\Dot;
+use StoreKeeper\WooCommerce\B2C\Exceptions\ShippingMethodImportException;
+use StoreKeeper\WooCommerce\B2C\Exceptions\WordpressException;
+use StoreKeeper\WooCommerce\B2C\I18N;
+use StoreKeeper\WooCommerce\B2C\Interfaces\WithConsoleProgressBarInterface;
+use StoreKeeper\WooCommerce\B2C\Models\ShippingMethodModel;
+use StoreKeeper\WooCommerce\B2C\Models\ShippingZoneModel;
+use StoreKeeper\WooCommerce\B2C\Traits\ConsoleProgressBarTrait;
+
+class ShippingMethodImport extends AbstractImport implements WithConsoleProgressBarInterface
+{
+    use ConsoleProgressBarTrait;
+
+    const SHIPPING_ZONE_NAME_PREFIX = 'SK_';
+    const SHIPPING_LOCATION_TYPE = 'country';
+
+    const SHIPPING_CLASS_FLAT_RATE = 'flat_rate';
+    const SHIPPING_CLASS_FREE_SHIPPING = 'free_shipping';
+    const SHIPPING_CLASS_LOCAL_PICKUP = 'local_pickup';
+
+    const FREE_SHIPPING_REQUIRES = 'min_amount';
+
+    const SK_SHIPPING_TYPE_ALIAS_PARCEL = 'Parcel';
+    const SK_SHIPPING_TYPE_ALIAS_TRUCK_DELIVERY = 'TruckDelivery';
+    const SK_SHIPPING_TYPE_ALIAS_PICKUP_AT_STORE = 'PickupAtStore';
+
+    const SK_SHIPPING_TYPE_MODULE = 'ShippingModule';
+
+    private int $storekeeper_id = 0;
+
+    public function __construct(array $settings = [])
+    {
+        $this->storekeeper_id = array_key_exists('storekeeper_id', $settings) ? (int) $settings['storekeeper_id'] : 0;
+        unset($settings['storekeeper_id']);
+        parent::__construct($settings);
+    }
+
+    protected function getModule()
+    {
+        return 'ShopModule';
+    }
+
+    protected function getFunction()
+    {
+        return 'listShippingMethodsForHooks';
+    }
+
+    protected function getFilters()
+    {
+        $filters = [
+            [
+                'name' => 'enabled__=',
+                'val' => '1',
+            ],
+            [
+                'name' => 'is_system__=',
+                'val' => '0',
+            ],
+        ];
+
+        if ($this->storekeeper_id > 0) {
+            $filters[] = [
+                'name' => 'id__=',
+                'val' => $this->storekeeper_id,
+            ];
+        }
+
+        return $filters;
+    }
+
+    protected function getLanguage()
+    {
+        return null;
+    }
+
+    /**
+     * @throws ShippingMethodImportException
+     * @throws WordpressException
+     */
+    protected function processItem(Dot $dotObject, array $options = [])
+    {
+        $this->debug('Processing shipping method', $dotObject->get());
+        $storekeeperId = $dotObject->get('id');
+
+        if ($this->isShippingTypeValid($dotObject)) {
+            $this->processShippingMethod($dotObject, $storekeeperId);
+        }
+
+        $this->debug("Unsupported shipping type {$dotObject->get('shipping_type.alias')}, will be skipped.", [
```

Comment:
the type can be changed in gui, so if it changes from know -> unknown we need to remove it

Reply:
Added handling for unknown types.